src/main/java/seedu/address/logic/commands/MarkCommand.java

```diff
@@ -21,43 +23,80 @@ public class MarkCommand extends Command {
     public static final String COMMAND_WORD = "mark";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Marks attendance for the contact "
-            + "by the index number used in the last person listing and for the tutorial number inputted. "
-            + "Parameters: INDEX (must be a positive integer) "
-            + "tut/TUTORIAL\n"
+            + "by the index number used in the last person listing and for the tutorial number inputted. \n"
+            + "Parameters: \n"
+            + "INDEX (must be a positive integer) \n"
+            + "tut/TUTORIAL in the format of \n"
+            + "1) A positive number between 1-12 \n"
+            + "2) A list of numbers eg. [1,3,5] \n"
+            + "3) A range of two numbers eg. 3-6 \n"
             + "Example: " + COMMAND_WORD + " 1 "
-            + "tut/1";
+            + "tut/1-5";
 
     private final Index index;
-    private final Tutorial tutorial;
+    private final List<Tutorial> tutorials;
 
     /**
      * @param index of the person in the display list
-     * @param tutorial number to mark attendance for
+     * @param tutorial single tutorial to mark attendance for
      */
     public MarkCommand(Index index, Tutorial tutorial) {
         this.index = index;
-        this.tutorial = tutorial;
+        this.tutorials = List.of(tutorial);
+    }
+
+    /**
+     * Constructor for marking multiple tutorials.
+     * @param index of the person in the display list
+     * @param tutorials list of tutorials to mark attendance for
+     */
+    public MarkCommand(Index index, List<Tutorial> tutorials) {
+        this.index = index;
+        this.tutorials = tutorials.stream().distinct().collect(Collectors.toList());
     }
 
     /**
      * @param personToEdit person whose attendance will be marked
-     * @param tutorial tutorial to mark attendance for
+     * @param tutorials tutorials to mark attendance for
      */
-    private Person generateMarkedPerson(Person personToEdit, Tutorial tutorial) throws CommandException {
+    private Pair<Person, String> generateMarkedPerson(Person personToEdit,
```

Comment:
Not sure if the method name "generateMarkedPerson" is appropriate here. Perhaps the logic of handling the unnecessary and markedTutorials could be abstracted out in next milestone